app/products/playbooks/screens/playbook_run/checklist/checklist_item/checklist_item_bottom_sheet.tsx

```diff
@@ -0,0 +1,251 @@
+        defaultMessage: 'Checked',
+    },
+    skip: {
+        id: 'playbooks.checklist_item.skip',
+        defaultMessage: 'Skip',
+    },
+    skipped: {
+        id: 'playbooks.checklist_item.skipped',
+        defaultMessage: 'Skipped',
+    },
+    runCommand: {
+        id: 'playbooks.checklist_item.run_command',
+        defaultMessage: 'Run command',
+    },
+    rerunCommand: {
+        id: 'playbooks.checklist_item.rerun_command',
+        defaultMessage: 'Rerun command',
+    },
+    assignee: {
+        id: 'playbooks.checklist_item.assignee',
+        defaultMessage: 'Assignee',
+    },
+    dueDate: {
+        id: 'playbooks.checklist_item.due_date',
+        defaultMessage: 'Due date',
+    },
+    command: {
+        id: 'playbooks.checklist_item.command',
+        defaultMessage: 'Command',
+    },
+    none: {
+        id: 'playbooks.checklist_item.none',
+        defaultMessage: 'None',
+    },
+});
+
+const ACTION_BUTTON_HEIGHT = 62;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => ({
+    scrollContentContainer: {
+        gap: 12,
+    },
+    container: {
+        flex: 1,
+        backgroundColor: theme.centerChannelBg,
+    },
+    checkboxContainer: {
+        flexDirection: 'row',
+        alignItems: 'flex-start',
+        gap: 12,
+    },
+    taskTitle: {
+        ...typography('Heading', 200, 'Regular'),
+        color: theme.centerChannelColor,
+    },
+    taskDescription: {
+        ...typography('Body', 100, 'Regular'),
+        color: changeOpacity(theme.centerChannelColor, 0.72),
+    },
+    actionButtonsContainer: {
+        flexDirection: 'row',
+        gap: 12,
+        height: ACTION_BUTTON_HEIGHT,
+    },
+    taskDetailsContainer: {
+        gap: 8,
+    },
+    flex: {
+        flex: 1,
+    },
+}));
+
+type Props = {
+    item: PlaybookChecklistItemModel | PlaybookChecklistItem;
+    assignee?: UserModel;
+    onCheck: () => void;
+    onSkip: () => void;
+    onRunCommand: () => void;
+    teammateNameDisplay: string;
+};
+
+const ChecklistItemBottomSheet = ({
+    item,
+    assignee,
+    onCheck,
+    onSkip,
+    onRunCommand,
+    teammateNameDisplay,
+}: Props) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+
+    const dueDate = 'dueDate' in item ? item.dueDate : item.due_date;
+    const isChecked = item.state === 'closed';
+    const isSkipped = item.state === 'skipped';
+    const isCommandRun = Boolean('commandLastRun' in item ? item.commandLastRun : item.command_last_run);
+
+    const handleCheck = useCallback(() => {
+        onCheck?.();
+        dismissBottomSheet();
+    }, [onCheck]);
+
+    const handleSkip = useCallback(() => {
+        onSkip?.();
+        dismissBottomSheet();
+    }, [onSkip]);
+
+    const handleRunCommand = useCallback(() => {
+        onRunCommand?.();
+        dismissBottomSheet();
+    }, [onRunCommand]);
```

Comment:
Shouldn't we dismiss the bottomSheet and await before executing the action? Or this does not affect with the potential snack bar (overlay screen) if there is an error?